### ISSUE TITLE
React Intl Integration

### ISSUE THREAD

**Issue by shenoudab** [reactjs/react-rails#227]
Dears, 

 Thanks for the amazing library for integrating reactjs with rails .. and making it easy to going for react in rails.

i was looking for formatting date and time and other formatings for react .. and i found React Intl
http://formatjs.io/react/

how to integrate React Intl in my rails app which use react-rails.

Thanks,
Shenouda Bertel

**Comment by blainekasten** [comment]
If you're using a build system, download the code through npm. Otherwise you'll probably need to copy/paste the code into a file and load it through the asset pipeline.

https://github.com/yahoo/react-intl/blob/master/dist/react-intl.js

**Comment by shenoudab** [comment]
Dear Blaine,

 I download react-intl through npm and adding it to my application.js file //= require react-intl and its loaded in header.

but when i add:
var IntlMixin     = ReactIntl.IntlMixin;
var FormattedDate = ReactIntl.FormattedDate;
to jsx file and adding mixins: [ReactIntl.IntlMixin], to React Class i got error "ReferenceError: Can't find variable: ReactIntl"

**Comment by blainekasten** [comment]
Sorry, maybe I wasn't as clear. 

Installing through npm doesn't give rails the natural ability to find the file. npm is helpful if you use an external build system integrated with rails (like webpack, grunt, gulp, browserify). I would suggest for your situation to simply copy the code, and paste it in 

```
vendor/
  javascripts/
    react-intl.js
```

That file will then be available to the asset pipeline.

**Comment by shenoudab** [comment]
Thanks Blaine, 

 for your fast reply .. its  placed the react-intl.js inside vendor/javascripts as your recommendations.

but i got the same error again .. i think the error is about declare ReactIntl not in the assets pipleline.

"ReferenceError: Can't find variable: ReactIntl"

as i add the following:
var IntlMixin     = ReactIntl.IntlMixin;
var FormattedDate = ReactIntl.FormattedDate;

it can't find a declare for ReactIntl.

Any other recommendations or suggestions ?

Thanks

**Comment by brettjurgens** [comment]
@shenoudab 
put `//= require vendor/react-intl` in your application.js and it should add it to the asset pipeline

**Comment by pacarvalho** [comment]
Are there plans to make this available through the usual "config.react.addons = true"?

**Comment by rmosolgo** [comment]
No, I expect `config.react.addons` will only include these: https://facebook.github.io/react/docs/addons.html

**Comment by goldnuggets24** [comment]
I have been in a few cage matches in the past couple weeks while attempting to integrate 3rd party NPM components with my React/Rails build. I'm not going to tell you I did this correctly, but this is what worked for me. I placed the following at the top of the component I called react-intl in: 

`var ReactIntl = require('react-intl/lib/react-intl');
var IntlMixin = ReactIntl.IntlMixin;
var FormattedDate = ReactIntl.FormattedDate;`

Basically, Rails wasn't finding 'ReactIntl' by searching for it at the root of my node-modules folder. I had to redirect it over to the associated lib folder. I had to do something similar with Material-UI. Hope that helps!
